### PR TITLE
Fix _get_file_contents to return bytes and checking that a different …

### PR DESCRIPTION
**Pull request by abadger**
…file exists than it opens

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
dataloader.py  and plugins/lookup/ini.py

##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

**Comment by abadger** [comment]
@Yannig just an FYI: we determined that _get_file_contents() needs to return byte strings.  Updated code and added a docstring to make that clear.

File: lib/ansible/parsing/dataloader.py [hunk 1]
```diff
@@ -416,9 +425,6 @@ def get_real_file(self, file_path, decrypt=True):
         if not self.path_exists(b_file_path) or not self.is_file(b_file_path):
             raise AnsibleFileNotFound(file_name=file_path)
 
-        if not self._vault:
-            self._vault = VaultLib(b_password="")
-
         real_path = self.path_dwim(file_path)
```
Comment (bcoca) [inline lib/ansible/parsing/dataloader.py]: this now seems redundant and doing almsot same as above?

**Comment by Yannig** [comment]
Fine. Thanks for your feedback :)

**Comment by abadger** [comment]
Merged to devel

**Pull request merged**